### ISSUE TITLE
Quick search Issues

### ISSUE THREAD

**Issue by vovanmix** [BoostIO/Boostnote#1263]
Quick search currently has some important issues that make it completely not useful:
- includes results from removed notes (trash). It doesn't seem right.
- only searches by note name, and not by the body.
- `cmd+a` to select the whole search string is not working, and if I want to change the search string I have to either hold backspace, or close/open the quick search.

**Comment by fidelaznar** [comment]
In version 0.9.0 Mac only searches content, not by note name. This could be solved for example by searching the title if not content is found. I think this function lives in browser/search.js, compiled in main.js:

function findByWord(notes, block) {
	  var regExp = new RegExp(_lodash2.default.escapeRegExp(block), 'i');
	  return notes.filter(function (note) {
	    if (_lodash2.default.isArray(note.tags) && note.tags.some(function (_tag) {
	      return _tag.match(regExp);
	    })) {
	      return true;
	    }
	    if (note.type === 'SNIPPET_NOTE') {
	      return note.description.match(regExp);
	    } else if (note.type === 'MARKDOWN_NOTE') {
	      if (note.content.match(regExp)==null) {
			return note.title.match(regExp);
	      } else {
	      	return note.content.match(regExp);
	      }
	    }
	    return false;
	  });
	}

**Comment by bimlas** [comment]
Since the note title is taken from the body (the first line of it is) and searching works on it, I think this issue may be closed.

**Comment by bhrd** [comment]
@bimlas 
I "think" this issue occurs for imported notes, since - with notes imported from other tools, the title doesn't show up in the body of the note.
I'm facing the same problem that @fidelaznar mentions, but just on those notes I imported from another tool - NVALT
I compared the .CSON files as well, for an imported note and a boostnote-created note.
In the imported note, the title is only present under the `title` key.
With the boostnote-created note, the title is present under `title` and `content` sections.
